source/module_elecstate/elecstate_print.cpp

```diff
@@ -10,6 +10,78 @@
 #include "module_base/formatter.h"
 namespace elecstate
 {
+	/**
+	 * Notes on refactor of ESolver's functions
+	 * 
+	 * the print of SCF iteration on-the-fly information.
+	 * 1. Previously it is expected for nspin 1, 2, and 4, also with xc_type 3/5 or not, the information will organized in different ways.
+	 * This brings inconsistencies between patterns of print and make it hard to vectorize information.
+	 * 2. the function print_etot actually do two kinds of things, 1) print information into running_*.log, 2) print information onto 
+	 * screen. These two tasks are, in no way should be placed/implemented in one function directly
+	 * 3. there are information redundance: the istep of SCF can provide information determing whether print out the SCF iteration info.
+	 * table header or not, rather than dividing into two functions and hard code the format.
+	 * 
+	 * For nspin 1, print: ITER, ETOT, EDIFF, DRHO, TIME
+	 * 	   nspin 2, print: ITER, TMAG, AMAG, ETOT, EDIFF, DRHO, TIME
+	 * 	   nspin 4 with nlcc, print: ITER, TMAGX, TMAGY, TMAGZ, AMAG, ETOT, EDIFF, DRHO, TIME
+	 * xc type_id 3/5: DKIN 
+	 * 
+	 * Based on summary above, there are several groups of info:
+	 * 1. counting: ITER
+	 * 2. (optional) magnetization: TMAG or TMAGX-TMAGY-TMAGZ, AMAG
+	 * 3. energies: ETOT, EDIFF
+	 * 4. densities: DRHO, DKIN(optional)
+	 * 5. time: TIME
+	 */
+	void print_scf_iterinfo(const std::string& ks_solver, const int& istep, const int& witer,
```

Comment:
Did you use 'clang_format' to format the source files?

Reply:
> Did you use 'clang_format' to format the source files?

@mohanchen sorry I did not, my considerations are:
There are too many args that needed, thus to avoid confusion whoreads this function, I group args and place in different lines, for exarmple the first line corresponds to the table data "GE1", "DA1", or "CC615", the second line corresponds to "AMAG" and "TMAG", the third line corresponds to "EDIFF" and "ETOT", the fourth "DRHO" and "LDKIN", the last "TIME".
I know clang-format will simply make the param list as a vertiical table but not the one I expect.

Reply:
@kirk0830 This is indeed a problem by the current clang-format syntax. Would you help me finding a better choice of the following parameters for clang-format?

https://clang.llvm.org/docs/ClangFormatStyleOptions.html
AlignAfterOpenBracket 
BinPackArguments
BinPackParameters